### ISSUE TITLE
Hash functions in mxnet?

### ISSUE THREAD

**Issue by yifeim** [apache/incubator-mxnet#10319]
Hi,

The sparse demos used python default hash functions. However, in my experience, python hash can be quite unreliable between different python versions. Could we implement something similar to http://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.FeatureHasher.html in mxnet?

https://github.com/apache/incubator-mxnet/blob/e91ff1b81bf2b5f4536ffb5b20e6b12e3cc63725/example/sparse/wide_deep/data.py#L110

**Comment by anirudhacharya** [comment]
@nswamy @sandeep-krishnamurthy Please label this - "Feature Request".
